src/ucp/tag/offload.c

```diff
@@ -144,61 +191,40 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_tag_offload_unexp_rndv,
 {
     ucp_worker_t *worker        = arg;
     ucp_request_hdr_t *rndv_hdr = (ucp_request_hdr_t*)hdr;
-    ucp_ep_t *ep                = ucp_worker_get_reply_ep(worker, rndv_hdr->sender_uuid);
-    size_t len                  = sizeof(ucp_rndv_rts_hdr_t);
     ucp_rndv_rts_hdr_t *rts;
-    ucp_sw_rndv_hdr_t  *sw_rndv_hdr;
-    const uct_md_attr_t *md_attr;
+    size_t len;
     size_t rkey_size;
 
-    if (!ucp_ep_is_stub(ep) && rkey_buf) {
-        md_attr   = ucp_ep_md_attr(ep, ucp_ep_get_tag_lane(ep));
-        rkey_size = md_attr->rkey_packed_size;
-        len += rkey_size;
-    } else {
-        rkey_size = 0;
-    }
-
-    rts = ucs_alloca(len);
-
-
-    /* Fill RTS to emulate SW RNDV flow. */
-    rts->super.tag = stag;
-    rts->sreq      = *rndv_hdr;
-    rts->address   = remote_addr;
-
+    rkey_size = ucp_tag_offload_rkey_size(worker->context);
+    rts       = ucs_alloca(sizeof(*rts) + rkey_size); /* SW rndv req may also
+                                                         carry a key */
     if (remote_addr) {
-        rts->size  = length;
-        rts->flags = UCP_RNDV_RTS_FLAG_OFFLOAD;
-        if (rkey_size) {
-            memcpy(rts + 1, rkey_buf, rkey_size);
-            rts->flags |= UCP_RNDV_RTS_FLAG_PACKED_RKEY;
-        }
+        /* Unexpected tag offload RNDV */
+        ucp_tag_offload_fill_rts(rts, rndv_hdr, stag, remote_addr, length, 0);
+        len = ucp_tag_offload_copy_rkey(rts, (void*)rkey_buf, rkey_size) +
+              sizeof(*rts);
```

Comment:
indentation

Reply:
it is not a function argument, but the second addendum in the expresion